gobblin-service/src/main/java/org/apache/gobblin/service/modules/scheduler/GobblinServiceJobScheduler.java

```diff
@@ -176,7 +172,7 @@ public void run() {
       // Since we are going to change status to isActive=false, unschedule all flows
       List<Spec> specs = new ArrayList<>(this.scheduledFlowSpecs.values());
       for (Spec spec : specs) {
-        onDeleteSpec(spec.getUri(), spec.getVersion());
+        unscheduleSpec(spec.getUri(), spec.getVersion());
```

Comment:
why not have ` this.scheduledFlowSpecs.remove(deletedSpecURI.toString());` also?